### DISCUSSION TITLE
How to send Pi CPU Temperature to i2c so Nixie can turn on fan when needed?

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#875] [Q&A]
Can some please tell me the secret sauce for forwarding the Pi's CPU temperature as a sensor to control when the case fan will go on/off?

The Pi appears to have 2 i2c buses, and bus 1 has the Megabas & Relay Hat on it.  

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/2f92cbcb-16a0-432a-b6cc-98f0e452f7dd)

I  can read the CPU temperature:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/0b4fb6a3-e798-4051-bec8-61d5844bc958)

Thanks for you help!

**Comment by tagyoureit** [comment]
This can all be done in REM and is _somewhere_ in the wiki, but for the life of me can't find it now.  There is a generic device for reading the Pi cpu temp.  You can feed this to a relay to turn the fan on/off.

![2023-12-21_12-48-01](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/43f5945b-08af-454d-96ad-b45840f53c18)

![2023-12-21_12-52-34](https://github.com/tagyoureit/nodejs-poolController/assets/1646840/42e12800-eb69-4405-950a-6dcc0b27ab3a)

**Reply by FlaMike** [reply]
I figured there has to be a way to do it.  I'll keep digging around & will let you know if I find it.  If you suddenly recall where it is before I report success, I'd appreciate you posting the wiki link.  No hurry, especially on Xmas  weekend.  I'm probably a week or two away from heading to the pad, but I'm trying to build as complete a system as possible before mounting it.  Pad gymnastics with teeny tiny lose parts is a sure recipe for frustration, in my experience.  

I even ordered a pressure sensor & whitebox hat from Atlas just so I could build my complete Pi stack on the workbench.  REM Chem parts won't even get ordered until Nixie is up, running & happy.  Nevertheless, I'm trying to get everything ready for a simpler pad experience.

As always, thanks for your help.

Have a wonderful Xmas and amazing 2024!

**Reply by FlaMike** [reply]
I think I'm close, just by following the screenshots you posted.  When I add the feed to the generic device & click save, the screen greys out & when it comes  back a bit later, the feed portion goes away.  Could this be because I've not defined any relays on the Hat yet?  

BTW, REM does show the CPU temperature within a few degrees of the Pi command line, so I must have done *something* right.

**Reply by tagyoureit** [reply]
> Could this be because I've not defined any relays on the Hat yet?

Yes, the feeds need a specific device to be sent to.

Also, if you didn't know, if you load up REM on a non-Linux box and set up these devices it will mock up fake values every few seconds.  It's a good way to test out a dev environment before trying to work out all the kinks on your pad.

**Reply by FlaMike** [reply]
Got it!  Working on setting up the fan relay & then will go back to the generic device setup.  Exciting stuff!!!!

I am accessing the Nixie via a Linux box, but it's a dual boot beast.  I  can run windows & access REM from there.  Will that do the trick?  If I understand you comment correctly, I'm guessing that won't do the trick--I need to install REM on winders.  Either way, I'm getting there, little by little.

**Reply by FlaMike** [reply]
The relays I anticipate using are now configured and toggle when I click them in REM.  I'm running into a  wall--same problem when I try to setup the feeds tab for the CPU temperature; screen greys out when I save & the feed lost when it returns.  I may just be ahead of my headlights because the Pi is just sitting on my desk with nothing  attached to it at the moment.

I'll look around the wikis & cogitate a bit.  Still at least I can see the CPU temperature in REM.

**Reply by FlaMike** [reply]
@tagyoureit 

I think I have it!  I found a discussion that appears to have the solution I need.  https://github.com/tagyoureit/nodejs-poolController/discussions/405

In my case, I'm using relay 5 for the case fan, so I changed the return code slightly.  If I set the temperature to be around the current CPU temperature, the relay clicks on &  off :-)

I do have one other hopefully easy question...  Sitting on my desk where the room temperature is about 70 F, the Pi's CPU temperature is about 110 F.  What is a reasonable temperature for turning the case fan on/off?

Thanks again for  your help, patience, & fabulous software!!

Have a wonderful holiday weekend!

![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/a7537dfc-aa4f-4035-88ec-0c6e8598558f)
